src/mechanics/cell/Cell.h

```diff
@@ -41,6 +41,24 @@ class Cell : public CellInterface
         return gradient;
     }
 
+    //! @brief builds the hessian0 matrix
+    DofMatrix<double> Hessian0() override
+    {
+        DofMatrix<double> hessian0;
```

Comment:
Will this not simply call the default constructor of Eigen, possibly resulting in a nonzero initial value?

Reply:
This is kind of a special behavior of the `DofContainer` things. They allow - as I called it - [uninitialized addition](https://github.com/nutofem/nuto/blob/master/test/mechanics/nodes/DofMatrix.cpp#L38). So this line will create a DofMatrix<double> containing zero sized vectors that is resized when needed.

Reply:
Okay. Not the most intuitive behaviour though.

Reply:
If you can come up with a better solution, please let me know. The problem is that the cell does not know the size of this container, or how many dof types it contains. We could do sth like
~~~c++
DofMatrix<double> hessian0 = mIntegrands[0].Gradient(...);
for (i = 1, ...) 
   hessian0 += mIntegrands[i].Gradient(....);
~~~
this seemed kind of silly though.